src/main/kotlin/resource/UserResource.kt

```diff
@@ -34,6 +57,76 @@ fun Route.user(userService: UserService) {
                 call.respond("You are missing one or multiple parameters")
             }
         }
+
+        webSocket("/balance") {
+            try {
+                val user = TokenManager.verifyTokenAndRetrieveUser(call.parameters)
+                val balanceManager = userService.buildBalanceManager(user)
+                val balanceIncreaseRateManager = BalanceIncreaseRateManager(user)
+
+                SessionSemaphore.acquireBalanceSession(user)
+
+                try {
```

Comment:
Method logic is hard to understand. I would suggest to refactor everything inside this try statement (including the try itself) to a separate method having a descriptive name.

Reply:
That's difficult to do here since I need the session information and the scope to send messages back onto the socket. I'll think about a solution regardless!